test/alternator/test_cql_rbac.py

```diff
@@ -124,3 +125,140 @@ def test_login_false(dynamodb, cql):
         with new_dynamodb(dynamodb, role, key) as d:
             with pytest.raises(ClientError, match='UnrecognizedClientException.*login=false'):
                 d.meta.client.list_tables()
+
+# Quote an identifier if it needs to be double-quoted in CQL. Quoting is
+# *not* needed if the identifier matches [a-z][a-z0-9_]*, otherwise it does.
+# double-quotes ('"') in the string are doubled.
+def maybe_quote(identifier):
+    if re.match('^[a-z][a-z0-9_]*$', identifier):
+        return identifier
+    return '"' + identifier.replace('"', '""') + '"'
+
+# Currently, some time may pass after calling GRANT or REVOKE until this
+# change actually takes affect: There can be group0 delays as well as caching
+# of the permissions up to permissions_validity_in_ms milliseconds.
+# This is why we need authorized() and unauthorized() in tests below - these
+# functions will retry the operation until it's authorized or not authorized.
+# To make tests fast, the permissions_validity_in_ms parameter should
+# be configured (e.g. test/cql-pytest/run.py) to be as low as possible.
+# But these tests should handle any configured value, as authorized() and
+# unauthorized() use exponential backoff until a long timeout.
+#
+# However, note that the long timeout means that a *failing* test will take
+# a long time. This is a big problem for xfailing tests, so those should
+# explicitly set timeout to a multiple of permissions_validitity_in_ms()
+# (see below).
+def authorized(fun, timeout=10):
```

Comment:
1) You can use a group0 barrier - there is now a rest API for it, or just issuing a DDL against a non-existing keyspace will do it for you.
2) Re authorization cache: @nuivall is working on an immediate propagation of the authorization information to the cache. It is unfortunate this patch didn't get in first. But perhaps in the meantime you could add a rest API to flush the cache?

Reply:
Unfortunately, as you noted the "group0 barrier" is only half of the story, and there is also the caching of the outdated information. Adding an API to flush the entire cache is one option, but an even better option would be to ensure that part of the RAFT operation is also to evict the outdated information (but... when reading we also need to ensure this node is up-to-date and I'm not at all sure how it's possible, since Raft only ensures a majority, not that every node got updated).

In any case, this test needed to work **now**. If in the future we improve the APIs that these tests have to work with - hopefully "GRANT"/"REVOKE" will just work and wait long enough for the following instructions to see the updated permissions - then we can simply this test (as well as other tests we already had for ACLs)